app/Event.php

```diff
@@ -25,6 +26,24 @@ public function contact()
         return $this->belongsTo('App\Contact');
     }
 
+    /**
+     * Limits the results to a specific object
+     *
+     * @param Builder $query
+     * @param Model $object
+     * @param string|null $key
+     * @return Builder
+     */
+    public function scopeForObject(Builder $query, Model $object, string $key = null)
```

Comment:
If you look at the destroy methods for the new controllers, I'm using this scope to find related objects more easily and erase them in one go. Like:

$contact->events()->forObject($activity)->get()

For standard event object names (kid, activity, etc) just lowercasing the model name works, so I don't have to explicitly pass the event object type this way and the method handles it.